pkg/gitreceive/k8s_util.go

```diff
@@ -0,0 +1,69 @@
+package gitreceive
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pborman/uuid"
+	"k8s.io/kubernetes/pkg/api"
+	utilyaml "k8s.io/kubernetes/pkg/util/yaml"
+)
+
+const (
+	v1Version = "v1"
+)
+
+func dockerBuilderPodName(appName, shortSha string) string {
+	uid := uuid.New()[:8]
+	return fmt.Sprintf("dockerbuild-%s-%s-%s", appName, shortSha, uid)
+}
+
+func slugBuilderPodName(appName, shortSha string) string {
+	uid := uuid.New()[:8]
+	return fmt.Sprintf("slugbuild-%s-%s-%s", appName, shortSha, uid)
+}
+
+func dockerBuilderPod(debug, withAuth bool, name, namespace string, env map[string]interface{}, tarURL, imageName string) *api.Pod {
+	pod := buildPod("dockerbuilder", withAuth, name, namespace, env)
+	return &pod
+}
+
+func slugbuilderPod(debug, withAuth bool, name, namespace string, env map[string]interface{}, tarURL, putURL, buildpackURL string) *api.Pod {
+	pod := buildPod("slugbuilder", withAuth, name, namespace, env)
+	return &pod
+}
+
+func buildPod(buildType string, withAuth bool, name, namespace string, env map[string]interface{}) api.Pod {
+	useCreds := "-no-creds"
+	if withAuth {
+		useCreds = ""
+	}
+	fileName := fmt.Sprintf("/etc/deis-%v%v.yaml", buildType, useCreds)
```

Comment:
can we completely remove the on-disk templates and instead construct them entirely in memory?